### PR TITLE
Enable react/no-array-index-key rule

### LINKED ISSUE #4814: ESLint: Prevent usage of Array index in keys
## Feature Description
We are currently using index as array keys in multiple places throughout the code, this, however, can potentially cause unexpected issues / unnecessary renders since an index as key behaves in a similar way as no key even though it doesn't display a warning. 

[See here](https://egghead.io/lessons/react-use-the-key-prop-when-rendering-a-list-with-react) in case you're interested in seeing a video comparing using no key vs index as key vs proper unique key.

The eslint rule: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md (Thanks @barklund for pointing out the rule exists)

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Testing Instructions

---

Fixes #4814

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -104,6 +104,7 @@
     "react/forbid-component-props": ["error", {
       "forbid": ["for"]
     }],
+    "react/no-array-index-key": "error",
```
Comment (swissspidy) [inline .eslintrc]:
```suggestion
    "react/no-array-index-key": "warn",
```

Let's make it a warning for now so we can already merge this and get the other pods to fix the issues.

Once that's done we can turn it into an error again.

Comment (spacedmonkey) [reply .eslintrc]: Agreed!

File: assets/src/dashboard/components/cardGallery/index.js [hunk 2]
```diff
@@ -153,7 +153,7 @@ function CardGallery({ story, isRTL, galleryLabel }) {
             const isActive = isCurrentPage && isInteractive;
             return (
               <ItemContainer
-                key={`page-${index}`}
+                key={page.id}
```
Comment (spacedmonkey) [inline assets/src/dashboard/components/cardGallery/index.js:156]: I think we need to prefix this. There will a couple of places we use page id as the key.

File: assets/src/dashboard/components/popoverPanel/index.js [hunk 3]
```diff
@@ -127,11 +127,11 @@ const PopoverPanel = ({
           <PillFieldset data-testid={'pill-fieldset'}>
             <Legend title={`options for ${title}`} />
             {items.map(
-              ({ label, selected, value, hex, disabled = false }, index) => {
+              ({ label, selected, value, hex, disabled = false }) => {
                 return (
                   <Pill
                     data-testid={'popover-pill'}
-                    key={`${value}_${index}`}
+                    key={value}
```
Comment (spacedmonkey) [inline assets/src/dashboard/components/popoverPanel/index.js]: Could this value not be unique here?

File: assets/src/i18n/translateWithMarkup.js [hunk 4]
```diff
@@ -121,6 +121,7 @@ function TranslateWithMarkup({ mapping = {}, children }) {
   return transform(
     new DOMParser().parseFromString(children, 'text/html').body.firstChild,
     mapping
+    //eslint-disable-next-line react/no-array-index-key
   ).map((element, index) => <Fragment key={index}>{element}</Fragment>);
```
Comment (spacedmonkey) [inline assets/src/i18n/translateWithMarkup.js:125]: Could we just a `uuid` here?

File: assets/src/dashboard/components/cardGallery/index.js (hunk 2 above)
Comment (swissspidy) [reply assets/src/dashboard/components/cardGallery/index.js:156]: Nope. The key only needs to be unique for this particular sub-tree, not for the whole app.

File: assets/src/i18n/translateWithMarkup.js (hunk 4 above)
Comment (swissspidy) [reply assets/src/i18n/translateWithMarkup.js:125]: Not needed. The size and order of this array is not changing, so it's OK to use the array index in this case.

File: assets/src/dashboard/components/popoverPanel/index.js (hunk 3 above)
Comment (swissspidy) [reply assets/src/dashboard/components/popoverPanel/index.js]: `value` is used for filtering in places like this, so it needs to be unique to make that work.

Screenshot 2020-10-27 at 10 59 35

File: assets/src/edit-story/components/colorPicker/gradientLine.js [hunk 5]
```diff
@@ -107,7 +107,7 @@ function GradientLine({
       {stops.map(({ position }, index) => (
         <GradientStop
           ref={(ref) => (stopRefs[index].current = ref)}
-          key={index}
+          key={position}
```
Comment (swissspidy) [inline assets/src/edit-story/components/colorPicker/gradientLine.js]: Not sure if this change is causing that one test failure... 🤔

**Review by swissspidy — APPROVED** [review]

**Pull request merged**